Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Indexer/NodeIndexer.php

```diff
@@ -277,7 +239,102 @@ public function indexNode(NodeInterface $node, $targetWorkspaceName = null)
+                        'script' => [
+                            'inline' => '
+                                    fulltext = (ctx._source.containsKey("__fulltext") ? ctx._source.__fulltext : new LinkedHashMap());
+                                    fulltextParts = (ctx._source.containsKey("__fulltextParts") ? ctx._source.__fulltextParts : new LinkedHashMap());
+                                    ctx._source = newData;
+                                    ctx._source.__fulltext = fulltext;
+                                    ctx._source.__fulltextParts = fulltextParts
+                                ',
+                            'params' => [
+                                'newData' => $documentData
+                            ]
+                        ],
+                        'upsert' => $documentData,
+                        'lang' => 'groovy'
+                    ]
+                ];
+            } else {
+                // non-fulltext-root documents can be indexed as-they-are
+                $this->currentBulkRequest[] = array(
+                    array(
+                        'index' => array(
+                            '_type' => $document->getType()->getName(),
+                            '_id' => $document->getId()
+                        )
+                    ),
+                    $documentData
+                );
+            }
+
+            $this->updateFulltext($node, $fulltextIndexOfNode, $targetWorkspaceName);
+        }
+    }
 
+    /**
+     * Remove document with the same contextPathHash but different NodeType, required after NodeType change
+     *
+     * @param string $contextPath
+     * @param string $contextPathHash
+     * @param NodeInterface $node
+     */
+    protected function removeDuplicateDocuments($contextPath, $contextPathHash, NodeInterface $node)
+    {
+        $type = NodeTypeMappingBuilder::convertNodeTypeNameToMappingName($node->getNodeType()->getName());
+        $this->logger->log(sprintf('NodeIndexer: Check duplicate nodes for %s (%s). ContentContextHash: %s', $contextPath, $type, $contextPathHash), LOG_DEBUG, null, 'ElasticSearch (CR)');
+        $result = $this->getIndex()->request('GET', '/_search?scroll=1m&search_type=scan', [], json_encode([
+            'query' => [
+                'bool' => [
+                    'must' => [
+                        'ids' => [
+                            'values' => [$contextPathHash]
+                        ]
+                    ],
+                    'must_not' => [
+                        'term' => [
+                            '_type' => $type
+                        ]
+                    ]
+                ]
+            ]
+        ]));
+        $treatedContent = $result->getTreatedContent();
+        $scrollId = $treatedContent['_scroll_id'];
+
+        $result = $this->getIndex()->request('GET', '/_search/scroll?scroll=1m', [], $scrollId, false);
+        $treatedContent = $result->getTreatedContent();
```

Comment:
The first treatedContent contains the scroll_id, the second the "real" treatedContent